### DISCUSSION TITLE
Initial connection a little slow. Rest is okay.

### DISCUSSION THREAD

**Discussion by jtnydv** [angristan/openvpn-install#836] [Help]
**Describe the issue**
Initial connection is taking very long to setup. Rest of the connection speed, once setup is good. However, when I first connect to the server it takes a while to come up i.e. no connection works, DNS is not resolved, nothing is working.

**Expected behavior**
I downloaded the script 3-4 months back and then it was perfectly okay. However, I did reinstall it today and this initial connection problem is coming up.

**Server if applicable):**

- OS: Debian 9.13
- Hosting provider (if applicable): Proxmox (Self-Hosted)

**Client (if applicable):**

- Device: Windows Machine
- OS: Windows 10 19041,685
- Client: OpenVPN Connect 3.2.2 (1455)

**Comment by angristan** [comment]
Not sure what could cause this, maybe the server doesn't have enough entropy so the handshake takes some time?

**Comment by TinCanTech** [comment]
> Client: OpenVPN Connect 3.2.2 (1455)

You could try Openvpn-2.5 on your client.
https://openvpn.net/community-downloads/
